metricq/history_client.py

```diff
@@ -240,9 +242,9 @@ async def stop(self, exception: Optional[Exception] = None):
     async def history_data_request(
         self,
         metric: str,
-        start_time: Timestamp,
-        end_time: Timestamp,
-        interval_max: Timedelta,
+        start_time: Optional[Timestamp],
+        end_time: Optional[Timestamp],
+        interval_max: Optional[Timedelta],
         request_type: HistoryRequestType = HistoryRequestType.AGGREGATE_TIMELINE,
```

Comment:
Shouldn't this be

```suggestion
        request_type: history_pb2.HistoryRequest.RequestTypeValue = HistoryRequestType.AGGREGATE_TIMELINE,
```

Then you also don't need `cast`.

Reply:
I think this will lead to confusion why one should use `HistoryRequestType.`... for this argument

Reply:
Well, I'm confused because I think using `HistoryRequestType` is just wrong. Not sure if we can and should do some `typing` magic that actually makes `isinstance(HistoryRequestType.AGGREGATE_TIMELINE, HistoryRequestType) == True`.

Reply:
`cast` only affects the type checker. `isinstance(HistoryRequestType.AGGREGATE_TIMELINE, HistoryRequestType)` is still `False`.

But we could remove `HistoryRequestType` and use `history_pb2.HistoryRequest.AGGREGATE_TIMELINE` etc. instead.

Reply:
Well, `HistoryRequestType` is just a shorthand. And it's also not a Type in the sense of a python type but in the other sense... And `type(HistoryRequestType.<...>) == history_pb2.HistoryRequest.RequestTypeValue`, right?

Reply:
`history_pb2.HistoryRequest.RequestTypeValue` is just a cast from int (`typing___cast(HistoryRequest.RequestTypeValue, 0)`) generated by the mypy-protobuf package

Reply:
See 52008fc. Normal python won't use the pyi file, so `history_pb2.HistoryRequest.RequestTypeValue` is not defined and an `AttributeError` is raised.